fix: 🐛 convert NAs to FALSE at end in prep for classifying

Aastedet opened the pull request:
## Description

- Fixes error in `join_inclusions()`:
   - Propagates values and handles `NA`'s correctly across the `has_*` columns after joining

Now passes all tests, including correctly in/excluding/classifying all 23 edge_cases. Note that two tests are skipped (the nc-case tests (fixed are underway in #366 ), and the arrow-backend test (#380 )). We should create issues on these.

Closes a bunch of issues that haven't been posted. Fixes the issue in draft-PR #378 (can be closed after this PR is merged).

While the pipeline works as intended now, there is room to improve `duckplyr` support/compatibility to increase performance #381

## Checklist

- [x] Ran `just run-all`

lwjohnst86 commented:
@Aastedet merging PRs is delayed by the size of the PR, since any unresolved comment blocks all the PR from being merged. In this case, if you had made four PRs, two of them would already be merged. Plus it makes it much easier for the reviewer to review smaller PRs. So next time, please ensure the PR is small and importantly *atomic* (a single conceptual change).

lwjohnst86 commented:
Also, the bugs you fix here are not related to #366. I tried your fix in that PR and it did nothing. They are also bugs at different stages of the algorithm. Here is in the middle, in that PR it is at the end.

signekb commented:
I have to agree with Luke here; even though the individual conceptual changes wouldn't have made the tests pass, having one conceptual change per PR makes the review process a lot smoother and focused. 

I must admit that I haven't reviewed this PR yet, mostly my focus has been on the registers2parquet package this week, but also bc the PR is so complex to me that I almost don't know where to start. 

As we've agreed, next week we'll all focus on this package. As I see it, we can move forward in the following ways: 

1) Anders makes four new PRs, all with one conceptual change, so it's easier to review and discuss.
2) We have a meeting about this PR on Monday and go through each of the conceptual changes together and take it from there. 
3) We keep discussing all the conceptual changes in this PR here (I'll enter the discussion/review this on Monday)

Thoughts, @Aastedet @lwjohnst86 ?

lwjohnst86 commented:
Good question. I think it would be better to split it up so the at least two of the changes can be merged in. Then the two other changes can be discussed and resolved separately (and not be dependent on them both being resolved in order to merge in).

lwjohnst86 commented on R/drop.R:
```diff
@@ -139,23 +139,20 @@ drop_pregnancies <- function(
+      .data$atc,
+      .data$apk,
+      .data$is_hba1c
+    )
```
Split this out into another PR :relaxed:

Aastedet commented:
I've updated the fix to `join_inclusions()` to use the `coalesce()` implementation.
I've also updated `test_classify_diabetes()` to handle the changes to it coming from the last merge from main (where, after switching to DuckDB, the edge_case dataset was mixed with the generated fake data to increase data size).

Now `classify_diabetes()` handles all the cases correctly again and passes the test. The only test failure is on the nc dataset which is currently bugged, so I've skipped it.

lwjohnst86 commented on R/join-inclusions.R:
```diff
@@ -36,5 +36,14 @@ join_inclusions <- function(
+    dplyr::mutate(
+      dplyr::across(
+        dplyr::starts_with("has_"),
+        ~ dplyr::coalesce(any(.x), FALSE)
```
Coalesce is a better choice since that's what the SQL term uses to describe handling NULL/NA values.
